crates/sui-benchmark/src/workloads/workload.rs

```diff
@@ -23,6 +29,57 @@ pub const STORAGE_COST_PER_COUNTER: u64 = 341 * 76 * 100;
 /// Used to estimate the budget required for each transaction.
 pub const ESTIMATED_COMPUTATION_COST: u64 = 1_000_000;
 
+#[derive(Debug, EnumCountMacro, EnumIter, Clone, Copy)]
+pub enum ExpectedFailureType {
+    Random = 0,
+    InvalidSignature,
+    // TODO: Add other failure types
+}
+
+impl TryFrom<u32> for ExpectedFailureType {
+    type Error = anyhow::Error;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(rand::random()),
+            _ => ExpectedFailureType::iter()
```

Comment:
The logic here just converts from an int to the Enum variant... the 0 is a special case because we pick a random variant. All other cases are handled the same though - just get the nth value of the iterator across all variants. So no need to match specifically against `InvalidSignature` variant.